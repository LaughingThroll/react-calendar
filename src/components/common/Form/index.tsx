import React from 'react'
import styles from './form.module.scss'

export interface FormProps {
  children?: any
  classNames?: string[]
}

const { form, form__item } = styles

const Form: React.FC<FormProps> = ({ children, classNames = [] }) => {
  return (
    <form className={[form, ...classNames].join(' ')}>
      {React.Children.map(children, (child, i) => (
        <div key={i} className={form__item}>
          {child}
        </div>
      ))}
    </form>
  )
}

export default React.memo(Form)
