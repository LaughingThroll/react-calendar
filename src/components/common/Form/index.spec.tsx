import { render } from 'enzyme'
import Form from '.'

describe('Form render', () => {
  it('render default', () => {
    const tree = render(<Form />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(0)
  })

  it('render with one child', () => {
    const tree = render(<Form children={<span>Test</span>} />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(1)
  })

  it('render with children', () => {
    const tree = render(
      <Form children={[<span key="1">Test</span>, <span key="2">Test</span>, <span key="3">Test</span>]} />
    )

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(3)
  })
})
