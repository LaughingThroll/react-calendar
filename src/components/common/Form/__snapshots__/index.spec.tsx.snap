// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render render default 1`] = `
<form
  class="form"
/>
`;

exports[`Form render render with children 1`] = `
<form
  class="form"
>
  <div
    class="form__item"
  >
    <span>
      Test
    </span>
  </div>
  <div
    class="form__item"
  >
    <span>
      Test
    </span>
  </div>
  <div
    class="form__item"
  >
    <span>
      Test
    </span>
  </div>
</form>
`;

exports[`Form render render with one child 1`] = `
<form
  class="form"
>
  <div
    class="form__item"
  >
    <span>
      Test
    </span>
  </div>
</form>
`;
